static/openapi/hss-spec.json

```diff
@@ -0,0 +1,3701 @@
+                        "title": "Enable",
+                        "description": "Indicates whether to enable extracting. If set to `False`, extracting will be disabled, and other extract configurations will be ignored.",
+                        "default": false
+                    },
+                    "logic": {
+                        "type": "string",
+                        "title": "Logic",
+                        "description": "JavaScript based data extracting function logic, accepting a `data` argument and returning either a JSON object or a string.Please note: The type of the `data` argument will depend on the ingested data. If the ingested data is structured (JSON or CSV), `data` will be a map/dictionary type. If the ingested data is unstructured (text, PDF, etc.), `data` will be an array of strings.",
+                        "default": "function process(data) { return data; }",
+                        "examples": [
+                            "function process(data) { return data; }"
+                        ]
+                    }
+                },
+                "type": "object",
+                "title": "ExtractConfig"
+            },
+            "FilterCondition": {
+                "type": "string",
+                "enum": [
+                    "AND",
+                    "OR"
+                ],
+                "title": "FilterCondition"
+            },
+            "FilterOperation": {
+                "type": "string",
+                "enum": [
+                    "==",
+                    "!=",
+                    "<",
+                    "<=",
+                    ">",
+                    ">=",
+                    "IN",
+                    "NOT_IN",
+                    "LIKE",
+                    "=~",
+                    "!~"
+                ],
+                "title": "FilterOperation"
+            },
+            "FlatIndexType": {
+                "properties": {
+                    "type": {
+                        "type": "string",
+                        "enum": [
+                            "flat"
+                        ],
+                        "const": "flat",
+                        "title": "Type",
+                        "description": "Type of the vector index.",
+                        "default": "flat"
+                    }
+                },
+                "type": "object",
+                "title": "FlatIndexType",
+                "description": "A Flat index, also known as a brute-force index, involves comparing the query vector directly against\nevery vector in the dataset to find the most similar items. While this method is simple and can be very\naccurate since it checks every possibility, it's also computationally expensive and slow, especially as\nthe dataset size grows. It is best suited for smaller datasets or scenarios where maximum accuracy is\nrequired and computational resources or time are not limiting factors."
```

Comment:
```suggestion
                "description": "A flat index, also known as a brute-force index, compares the query vector directly against\nevery vector in the dataset to find the most similar items. While this method is simple and can be very\naccurate since it checks every possibility, it's also computationally expensive and slow, especially as\nthe dataset size grows. It is best suited for smaller datasets or scenarios where maximum accuracy is\nrequired and computational resources or time are not limiting factors."
```